addons/mail/models/mail_channel.py

```diff
@@ -282,6 +282,13 @@ def write(self, vals):
 
         if vals.get('group_ids'):
             self._subscribe_users()
+        # Reflect changes to every members when switching a channel to a mailing one or vice versa
+        if 'email_send' in vals:
+            notifications = []
+            for record in self:
+                for partner in record.channel_partner_ids:
+                    notifications.append([(self._cr.dbname, 'res.partner', partner.id), {'channel_id': record.id, 'type': 'toggle_mass_mailing'}])
+                self.env['bus.bus'].sendmany(notifications)
```

Comment:
Few remarks here:
- we don't need to send the notification if the value didn't change
- for channels, we want to send the notification once on the channel, and not for every partner
- I would move this at the end of the method instead of in the middle, just to make sure all computations are done before we send info
- It is very important to send the new value, because a toggle assumes the initial state is known, which is very prone to errors

Reply:
@seb-odoo . there are some clarification required.

Point 1 - Actually we have to toggle the channel type. so I have checked the "email_send" either it is true or false. we have to made the changes accordingly.
Point 2 - You mean, We have to make the bus notification for "mail.channel" instead "res.partner" ? actually in the task description, you have mentioned "should send a bus notification to channel members when channel type changes".

Reply:
Point 1:
Toggle means going from a supposedly known initial state, to its "inverse" value. In this case the JS might not even be aware of the initial state, so it's not possible to instruct "toggle".
The dict that you send on the bus should simply contain the `mass_mailing` key with it's new value.

(Btw even if the JS was aware of the initial state, "toggle" is a really bad instruction especially in async flows, because if the same instructions is sent multiple times and any of them is lost in the network, or received 2 times because our bus code is not very robust, the final state will be corrupted. Another possible scenario is 2 users are doing the change in parallel, so it might be sent 2 times even it should only be changed to one specific final state).

Point 2:
Correct, sorry if the spec was not clear, but "target channel members" meant sending it on the channel itself, this is the way we communicate with all its members at once.